tests/model_registry/conftest.py

```diff
@@ -298,8 +303,44 @@ def model_registry_instance_rest_endpoint(
 
 
 @pytest.fixture(scope="class")
-def generated_schema(model_registry_instance_rest_endpoint: str) -> Any:
-    return schemathesis.from_uri(
-        uri="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml",
-        base_url=f"https://{model_registry_instance_rest_endpoint}/",
+def generated_schema(model_registry_instance_rest_endpoint: str) -> BaseOpenAPISchema:
+    schema = schemathesis.openapi.from_url(
+        url="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml"
     )
+    schema.configure(base_url=f"https://{model_registry_instance_rest_endpoint}/")
+    return schema
+
+
+# Used when testing local changes
+# @pytest.fixture(scope="class")
+# def generated_schema_local(model_registry_instance_rest_endpoint):
+#     schema = schemathesis.openapi.from_path(
+#         path="utilities/manifests/model-registry-api.yaml",
+#     )
+#     schema.configure(base_url=f"https://{model_registry_instance_rest_endpoint}/")
+#     return schema
+
+
+@pytest.fixture
+def state_machine(generated_schema: BaseOpenAPISchema, current_client_token: str) -> APIStateMachine:
```

Comment:
this can be a regular function; there's no setup / teardown related to the test here

Reply:
Schemathesis' documentation recommends using a fixture when loading the schema lazily: https://schemathesis.readthedocs.io/en/stable/stateful.html#schemathesis.stateful.state_machine.APIStateMachine.validate_response